nomos-mix/network/src/behaviour.rs

```diff
@@ -11,21 +12,32 @@ use libp2p::{
     },
     Multiaddr, PeerId,
 };
+use nomos_mix::conn_maintenance::{ConnectionMaintenance, ConnectionMaintenanceSettings};
 use nomos_mix_message::MixMessage;
 use sha2::{Digest, Sha256};
 use std::marker::PhantomData;
 use std::{
     collections::{HashMap, HashSet, VecDeque},
+    pin::pin,
     task::{Context, Poll, Waker},
 };
 
 /// A [`NetworkBehaviour`]:
 /// - forwards messages to all connected peers with deduplication.
 /// - receives messages from all connected peers.
-pub struct Behaviour<M> {
-    config: Config,
+pub struct Behaviour<M, Interval> {
+    config: Config<Interval>,
     /// Peers that support the mix protocol, and their connection IDs
     negotiated_peers: HashMap<PeerId, HashSet<ConnectionId>>,
+    /// Connection maintenance
+    /// NOTE: For now, this is optional because this may close too many connections
+    /// until we clearly figure out the optimal parameters.
+    conn_maintenance: Option<ConnectionMaintenance<PeerId>>,
+    /// Peers that should be excluded from connection establishments
+    blacklist_peers: HashSet<PeerId>,
+    /// To maintain address of peers because libp2p API uses only [`PeerId`] when handling
+    /// connection events. But, we need to know [`Multiaddr`] of peers to dial to them.
+    peer_addresses: HashMap<PeerId, Multiaddr>,
```

Comment:
I think all this should be part of some abstracted tool instead of something embedded completely in the behaviour itself.